x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/management/trial_license_complete_tier/upgrade_review_prebuilt_rules.ts

```diff
@@ -0,0 +1,679 @@
+                createRuleAssetSavedObject({
+                  rule_id: 'rule-1',
+                  version: 2,
+                  name: 'B',
+                }),
+              ];
+              await createPrebuiltRuleAssetSavedObjects(es, updatedRuleAssetSavedObjects);
+
+              // Call the upgrade review prebuilt rules endpoint and check that one rule is eligible for update
+              // but does NOT contain single line string field
+              const reviewResponse = await reviewPrebuiltRulesToUpgrade(supertest);
+              expect(reviewResponse.rules[0].diff.fields).toMatchObject({
+                version: {
+                  current_version: 1,
+                  diff_outcome: ThreeWayDiffOutcome.StockValueCanUpdate,
+                  has_conflict: false,
+                  has_update: true,
+                  merge_outcome: ThreeWayMergeOutcome.Target,
+                  merged_version: 2,
+                  target_version: 2,
+                },
+              });
+              expect(reviewResponse.rules[0].diff.has_conflict).toBe(false);
+              expect(reviewResponse.stats.num_rules_to_upgrade_total).toBe(1);
+            });
+          });
+
+          describe('when rule field has an update and a custom value that are different', () => {
+            it('should show in the upgrade/_review API response', async () => {
+              // Install all prebuilt detection rules
+              const ruleAssetSavedObjects = getRuleAssetSavedObjects();
+              await createPrebuiltRuleAssetSavedObjects(es, ruleAssetSavedObjects);
+              await installPrebuiltRules(es, supertest);
+
+              // Clear previous rule assets
+              await deleteAllPrebuiltRuleAssets(es, log);
+
+              // Customize a single line string field on the installed rule
+              await patchRule(supertest, log, {
+                rule_id: 'rule-1',
+                name: 'B',
+              });
+
+              // Increment the version of the installed rule, update a single line string field, and create the new rule assets
+              const updatedRuleAssetSavedObjects = [
+                createRuleAssetSavedObject({
+                  rule_id: 'rule-1',
+                  version: 2,
+                  name: 'C',
+                }),
+              ];
+              await createPrebuiltRuleAssetSavedObjects(es, updatedRuleAssetSavedObjects);
+
+              // Call the upgrade review prebuilt rules endpoint and check that one rule is eligible for update
+              // and single line string field update does not have conflict
+              const reviewResponse = await reviewPrebuiltRulesToUpgrade(supertest);
+              expect(reviewResponse.rules[0].diff.fields).toMatchObject({
+                name: {
+                  current_version: 'B',
+                  diff_outcome: ThreeWayDiffOutcome.StockValueCanUpdate,
+                  has_conflict: false,
```

Comment:
Should this actually be `has_conflict: true`? The test plan mentions that there should be a conflict, and I think it makes sense. 

Let's you have a rule with a customized field value, then the base version gets removed somehow. In such case I think we shouldn't automatically choose the target version, but force the user to make this choice.

Reply:
Yeah I agree, I'll change the logic of what we're returning to match this output

Reply:
I think this poses another question actually. Currently, whenever the `base_version` is missing, we assume the rule is _not_ customized giving us the diff outcomes of `StockValueNoUpdate` for the `-AA` situation and `StockValueCanUpdate` for the `-AB` situation. [Here is where that logic occurs in the code](https://github.com/elastic/kibana/blob/d452a6228b731a9b9fbcd71399290fbadbbfe6f9/x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/model/diff/three_way_diff/three_way_diff_outcome.ts#L41)

Should we be instead assuming both _are_ customized instead and the fallback for an undefined `base_version` would be to return the field and let users decide if they want to keep or update the field? 

In this scenario we would return `CustomizedValueSameUpdate` for the now assumed `-BB` outcome and `CustomizedValueCanUpdate` for the `-BC` outcome. cc: @banderror @nikitaindik

Reply:
I think it's okay to return a bunch of fields in this edge case, since it should only happen once. Base version would get created if you upgrade a rule, right? Otherwise we run the risk of removing user's custom value with no way to return to it.

This was my line of thinking:
If we have the `-AA` situation this could mean one of:
 - `AAA` -> no conflict, pick target
 - `ABB` -> no conflict, pick target
Since we don't know which case it is, we can assume the most complex scenario (ABB) and return `ThreeWayDiffOutcome.CustomizedValueSameUpdate`. Then indeed we'll return every field in response.

If we have the `-AB` situation this could mean one of:
 - `AAB` -> no conflict, pick target
 - `ABC` -> has conflict, user has to choose
Again, I would assume the "worst" (ABC) and return `CustomizedValueCanUpdate`. Then the field would also have `has_conflict: true` and the user would have to make a choice.

Reply:
I think we can merge this as is for now and wait until folks come back from vacations to discuss with them. Looking at the comment [here](https://github.com/elastic/kibana/blob/d452a6228b731a9b9fbcd71399290fbadbbfe6f9/x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/model/diff/three_way_diff/three_way_diff_outcome.ts#L43), perhaps there was an idea behind the current behaviour 🤷 .

Reply:
Pinging @approksiu for awareness and feedback

Reply:
I put this logic (the 3rd, compromise option) into a PR here: https://github.com/elastic/kibana/pull/186435

Reply:
I have read the thread and agree with the compromise logic option. Does it make sense to inform the user about missing the base version when this happens?

Reply:
@approksiu I think it's probably a good idea, not sure if that's in the UI plans yet or not

Reply:
We'll discuss with @ARWNightingale.

Reply:
Thanks for the feedback @approksiu, we'll go with option 3 then.

> Does it make sense to inform the user about missing the base version when this happens?
> not sure if that's in the UI plans yet or not

It would be a rare edge case to handle. It's not planned. I think it might be useful eventually, especially when we reach that 15k limit, but at this point I would not include this into Milestone 3. I opened a https://github.com/elastic/kibana/issues/186880 that we can consider adding to Milestone 4.

cc @dplumlee @jpdjere @xcrzx @nikitaindik

Reply:
Agreed, definitely in Milestone 3. Thanks for adding the issue!